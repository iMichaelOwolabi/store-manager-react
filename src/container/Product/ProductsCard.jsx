import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Button, Card } from 'semantic-ui-react';

const extra = <Button type="submit" className="details">Details</Button>;

const ProductsCard = ({ products }) => {
  if (products) {
    return (
      products.map(product => (
        <Link to={`/products/${product.productid}`} key={product.productid}>
          <Card
          key={product.productid}
          className="products"
          image={product.productimage}
          header={product.productname}
          meta="The most amazing product money can buy"
          description={product.price}
          extra={extra}
        />
        </Link>
      ))
    );
  }
  return <h2 className="noProduct">There are presently no products in the store!</h2>;
};

ProductsCard.propTypes = {
  products: PropTypes.array
};
ProductsCard.defaultProps = {
  products: []
};

export default ProductsCard;
