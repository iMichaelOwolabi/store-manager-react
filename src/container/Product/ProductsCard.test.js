import React from 'react';
import { shallow } from 'enzyme';
import ProductsCard from '.';

describe('Test the Products Card Component', () => {
  it('should match snapshot', () => {
    const props = {
      products: [{}]
    };
    const wrapper = shallow(<ProductsCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
