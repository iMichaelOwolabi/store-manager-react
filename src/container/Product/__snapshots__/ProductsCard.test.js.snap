// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Products Card Component should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
